xdsl/ir/core.py

```diff
@@ -2593,9 +2607,18 @@ def clone_into(
             for op in block.ops:
                 new_block.add_op(
                     op.clone(
-                        value_mapper, block_mapper, clone_name_hints=clone_name_hints
+                        value_mapper,
+                        block_mapper,
+                        clone_name_hints=clone_name_hints,
+                        clone_operands=False,
                     )
                 )
+        # Handle graph regions, where results may be created after their first use
```

Comment:
I would change that comment, this is valid for all regions (since blocks are not ordered)